Makefile

```diff
@@ -37,92 +36,43 @@ CROSS_COMPILE_32 ?= arm-none-linux-gnueabihf-
-ifneq ($(SECURITY_TYPE),gp)
-	$(SECDEV_DIR)/scripts/secure-binary-image.sh $(O)/arm-trusted-firmware/k3/$(TFA_BOARD)/release/bl31.bin $(O)/arm-trusted-firmware/k3/$(TFA_BOARD)/release/bl31.bin$(HS_EXT)
-	$(Q)cp -v $(O)/arm-trusted-firmware/k3/$(TFA_BOARD)/release/bl31.bin$(HS_EXT) $(I)
-endif
 
 optee: $(O) $(I)
 	$(Q)$(MAKE) -C $(OPTEE_DIR) O=$(O)/optee CROSS_COMPILE=$(CROSS_COMPILE_32) CROSS_COMPILE64=$(CROSS_COMPILE_64) PLATFORM=$(OPTEE_PLATFORM) $(OPTEE_EXTRA_ARGS) CFG_TEE_CORE_LOG_LEVEL=2 CFG_TEE_CORE_DEBUG=y CFG_ARM64_core=y all
-	$(Q)cp -v $(O)/optee/core/tee-pager_v2.bin $(I)/
-ifneq ($(SECURITY_TYPE),gp)
-	$(SECDEV_DIR)/scripts/secure-binary-image.sh $(O)/optee/core/tee-pager_v2.bin $(O)/optee/core/tee-pager_v2.bin$(HS_EXT)
-	$(Q)cp -v $(O)/optee/core/tee-pager_v2.bin$(HS_EXT) $(I)/
-endif
-
-dm: $(I)
-ifneq ($(DM_COMBINED_WITH_TIFS),1)
-	$(Q)cp -v $(FW_DIR)/ti-dm/$(DM_SOC_NAME)/ipc_echo_testb_mcu1_0_release_strip.xer5f $(I)
-ifneq ($(SECURITY_TYPE),gp)
-	$(SECDEV_DIR)/scripts/secure-binary-image.sh $(I)/ipc_echo_testb_mcu1_0_release_strip.xer5f $(I)/ipc_echo_testb_mcu1_0_release_strip.xer5f$(HS_EXT)
-endif
-endif
-
-u_boot_r5: $(O) $(I)
-	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_32) ARCH=arm TI_SECURE_DEV_PKG=$(SECDEV_DIR) O=$(O)/u-boot/r5 $(UBOOT_ARMV7_DEFCONFIG)
-	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_32) ARCH=arm TI_SECURE_DEV_PKG=$(SECDEV_DIR) O=$(O)/u-boot/r5
-ifeq ($(MULTICERTIFICATE_BOOT_CAPABLE),1)
-	$(Q)cp -v $(O)/u-boot/r5/spl/u-boot-spl.bin $(I)
-else
-	$(Q)cp -v $(O)/u-boot/r5/tiboot3.bin $(D)
-endif
-
-u_boot_armv8: $(O) $(D) optee tfa dm
-	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_64) ARCH=arm TI_SECURE_DEV_PKG=$(SECDEV_DIR) O=$(O)/u-boot/armv8 $(UBOOT_ARMV8_DEFCONFIG)
-	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_64) ARCH=arm TI_SECURE_DEV_PKG=$(SECDEV_DIR) O=$(O)/u-boot/armv8 \
-					ATF=$(I)/bl31.bin$(HS_EXT) \
-				  TEE=$(I)/tee-pager_v2.bin$(HS_EXT) \
-				  $(DMCONF)
-	$(Q)cp -v $(O)/u-boot/armv8/tispl.bin$(HS_UBOOTEXT) $(D)/tispl.bin
-	$(Q)cp -v $(O)/u-boot/armv8/u-boot.img$(HS_UBOOTEXT) $(D)/u-boot.img
+	$(Q)cp -v $(O)/optee/core/tee-pager_v2.bin $(I)
+
+u_boot_r5: $(O) $(D)
+	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_32) ARCH=arm O=$(O)/u-boot/r5 $(UBOOT_ARMV7_DEFCONFIG)
+	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_32) ARCH=arm O=$(O)/u-boot/r5 BINMAN_INDIRS=$(FW_DIR)
+	$(Q)cp -v $(O)/u-boot/r5/tiboot3-$(SOC_NAME)-$(SECURITY_TYPE)-evm.bin $(D)/tiboot3.bin
+
+u_boot_armv8: $(O) $(D) optee tfa
+	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_64) ARCH=arm O=$(O)/u-boot/armv8 $(UBOOT_ARMV8_DEFCONFIG)
+	$(Q)$(MAKE) -C $(UBOOT_DIR) CROSS_COMPILE=$(CROSS_COMPILE_64) ARCH=arm O=$(O)/u-boot/armv8 BINMAN_INDIRS=$(FW_DIR) \
+					BL31=$(I)/bl31.bin \
+				  TEE=$(I)/tee-pager_v2.bin
+	$(Q)cp -v $(O)/u-boot/armv8/tispl.bin$(SECTYPE_EXT) $(D)/tispl.bin
```

Comment:
?? 
SECTYPE_EXT is defined as '_unsigned'.

Reply:
tispl.bin_unsigned is not a file produced by binman. So back to the original question, what version of u-boot are you using?

Reply:
yes, binman branch only works with the TI SDK trees.  That's the idea, right?  Once TI support for binman goes upstream, we will support from the main branch.

Reply:
Right, my point is in this commit you didn't actual redirect the .gitsubmodule file to use the TI SDK tree.

Reply:
So someone who clones this branch, the build will fail, you should update the u-boot project submodule in this same commit.

Reply:
oh, I assumed this repo builds upstream by default, so to build another repo, you need to define UBOOT_DIR on the makefile invocation.  And so to use the binman branch, since it doesn't work on upstream, you would always include UBOOT_DIR=<your ti uboot branch>.

I hadn't considered hardcoding it to git.ti.com, but we could.  Thoughts??

Reply:
The main branch should default to upstream, but for this binman testing branch it would be nice to have the default point to a version of u-boot with binman.